### PR TITLE
Add browser and browserContext type definitions for k6/experimental/browser

### PR DESCRIPTION
**Pull request by ankur22**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/k6/experimental/browser/index.d.ts [hunk 1]
```diff
@@ -387,3 +383,166 @@ export interface ElementStateFilter {
...
+
+    /**
+     * Whether to activate JavaScript support for the context. Defaults
+     * to `false`.
+     */
+    javaScriptEnabled?: boolean;
+
+    /**
+     * Specifies the user's locale, such as 'en-US', 'tr-TR', etc.
+     * Defaults to what is set on the system.
```
Comment (ka3de) [inline types/k6/experimental/browser/index.d.ts:481]: Small suggestion:
```suggestion
     * Specifies the user's locale following ICU locale (e.g. 'en_US').
     * Defaults to host system locale.
```

**Review by ka3de — COMMENTED** [review]
Thanks for adding this @ankur22 ! :clap: 
Overall it LGTM, just made a few small suggestions.

File: types/k6/experimental/browser/browser.d.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    ): Page;
+
+    /**
+     * Currently only detects the `disconnected` event from the browser
+     * application.
+     * @param event
+     */
+    on(
+        event: 'disconnected',
+    ): boolean;
```
[the remark this answers is not in the document]
Comment (ankur22) [reply types/k6/experimental/browser/browser.d.ts:60]: Right, good catch. I'll remove it from the type definitions 👍

[the remark this answers is not in the document]
Comment (ankur22) [reply types/k6/experimental/browser/browser.d.ts]: Removed in commit: 098b1da87213c06e4057f53f7b28c615077a6632

I'll only merge this all in after the next release.

**Review by ka3de — APPROVED** [review]
LGTM :rocket:

**Review by oleiade — APPROVED** [review]

**Comment by ankur22** [comment]
Closing since https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66111 has been merged in

**Comment by oleiade** [comment]
Should be merged with version `v0.46.0`

**Pull request closed without merging**